### PR TITLE
Refactor python client api for project libraries

### PR DESCRIPTION
**Pull request by ehysi**
https://app.shortcut.com/dataiku/story/97735/add-project-libs-management-in-python-client-api

### CODE REVIEW

File: dataikuapi/dss/projectlibrary.py [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+        self.children.add(child)
+
+    def _remove_child_(self, child):
+        """
+        Removes the given child from this folder
+        """
+        self.children.remove(child)
+
+
+class DSSLibraryFile(DSSLibraryItem):
```
Comment (MayeulRousselet) [inline dataikuapi/dss/projectlibrary.py:252]: Maybe we could store the content of the file in a prop, `content`
get_content()  -> do the call to get the file contents and update `content`
set_content() -> overwrite `content`
save() -> do the POST request to save the file 

WDYT?

File: dataikuapi/dss/projectlibrary.py [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+class DSSLibrary(object):
+    """
+    A handle to manage the library items of a project
+    It saves locally a copy of taxonomy to help navigate in the library
+    All modifications done through this object and related library items are done locally and on remote.
+
+    Note: Taxonomy modifications done outside this library are not reflected locally.
+          You should reload the library in this case.
+    """
+    def __init__(self, client, project_key):
```
Comment (MayeulRousselet) [inline dataikuapi/dss/projectlibrary.py:13]: I am not a fan of calling `reload` in the constructor.
For what I know, but I am not the biggest expert on the matter (You should ask guidance to Clement), usually in the client, the constructor will just init the handle, and do not perform any call. A subsequent call to `get_settings` will do it. cf what is done for the wiki.

**Review by MayeulRousselet — CHANGES_REQUESTED** [review]
Thanks for this good work Erin, I have developped in the review some of my interrogations.

File: dataikuapi/dss/projectlibrary.py [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+        for c in self.children:
+            if c.name == name:
+                return c
+        return None
+
+    def add_file(self, file_name):
+        """
+        Create a handle to a new file in the library folder
+
+        Note : It is still not saved in DSS.
```
Comment (MayeulRousselet) [inline dataikuapi/dss/projectlibrary.py:230]: Precise: it is not saved in DSS until `contents(...)` is called

**Review by MayeulRousselet — CHANGES_REQUESTED** [review]

**Review by MayeulRousselet — APPROVED** [review]
LGTM

**Comment by apichery** [comment]
Replaced by https://github.com/dataiku/dip/pull/18089

**Pull request closed without merging**